libcxx/include/stddef.h

```diff
@@ -36,16 +24,19 @@
 
 */
 
-#  include <__config>
+#include <__config>
+
+// Note: This include is outside of header guards because we sometimes get included multiple times
+//       with different defines and the underlying <stddef.h> will know how to deal with that.
+#include_next <stddef.h>
```

Comment:
Our builds now fail with

```
In file included from ../../base/allocator/partition_allocator/src\partition_alloc/partition_alloc_base/debug/alias.h:8:
In file included from ../../third_party/libc++/src/include\cstddef:42:
../../third_party/libc++/src/include\stddef.h(31,2): error: #include_next is a language extension [-Werror,-Wgnu-include-next]
   31 | #include_next <stddef.h>
      |  ^
1 error generated.
```

…presumably because this is now before the `#pragma GCC system_header` below. Was that an intentional change?

(https://ci.chromium.org/ui/p/chromium/builders/try/win-rel/560031/overview)

Reply:
Hmm, no that's not intentional. Is `-Wgnu-include-next` enabled by default? Why did we not see this anywhere else?

I can fix that but it would be nice to have a test that fails if we break this again in the future.

Reply:
https://github.com/llvm/llvm-project/pull/88214

Reply:
Thanks for the fix!

> Is -Wgnu-include-next enabled by default?

No. Apparently someone over here thought it'd be good to compile some of our files with `-pedantic`, which enables `-Wgnu`.

> Why did we not see this anywhere else?

I don't know how libc++ tests run. Maybe there isn't a test that includes all public headers with `-Wall -Wconversion -Wextra -pedantic -Werror`? Does libc++ have any policy which warning levels it's clean at? Probably not really needed since all its headers are supposed to be system headers which disable warnings, so the test could and probably should even use -Weverything with clang.